cclib/method/volume.py

```diff
@@ -149,53 +159,95 @@ def getbfs(coords, gbasis):
         }
 
     bfs = []
-    for i, atom in enumerate(mymol):
-        bs = gbasis[i]
-        for sym, prims in bs:
-            for power in sym2powerlist[sym]:
-                bf = CGBF(atom.pos(), power)
-                for expnt, coef in prims:
-                    bf.add_primitive(expnt, coef)
-                bf.normalize()
-                bfs.append(bf)
-
+    
+    #PyQuante
+    if _found_pyquante:
+        for i, atom in enumerate(pymol):
+            bs = ccmol.gbasis[i]
+            for sym, prims in bs:
+                for power in sym2powerlist[sym]:
+                    bf = CGBF(atom.pos(), power)
+                    for expnt, coef in prims:
+                        bf.add_primitive(expnt, coef)
+                    bf.normalize()
+                    bfs.append(bf)
+
+    #pyquante2
+    if _found_pyquante2:
+        for i, atom in enumerate(pymol): # `atom` is instance of pyquante2.geo.atom.atom class.
+            basis = ccmol.gbasis[i] # `basis` is basis coefficients stored in ccData.
+            for sym, primitives in basis:
+                for power in sym2powerlist[sym]: # `sym` is S, P, D, F and is used as key here.
+                    exponentlist = []
+                    coefficientlist = []
+                    
+                    for exponents, coefficients in primitives:
+                        exponentlist.append(exponents)
+                        coefficientlist.append(coefficients)
+                    
+                    basisfunction = cgbf(atom.atuple()[1:4], exps=exponentlist, coefs=coefficientlist)
+                    basisfunction.normalize()
+                    bfs.append(basisfunction)
+                    
     return bfs
 
+# Small wrapper PyQuante & pyquante2 function that evaluates basis function on a given point
+# Used in both `wavefunction` and `electrondensity`
+def pyamp(bfs, bs, x, y, z):
+    if _found_pyquante2:

```

Comment:
What if both of these are false?

Reply:
If both are false, `pyamp` wouldn't be called by `wavefunction` or `electrondensity` because each of the function's first lines -- getbfs include checking the existence of the packages and would throw Exception if any of pyquante or pyquante2 is not present. I could add the exception here as well if that would make things more explicit?